jekyll/_cci2/rerun-failed-tests-only.adoc

```diff
@@ -0,0 +1,275 @@
+---
+contentTags:
+  platform:
+  - Cloud
+---
+= Re-run failed tests only
+:page-layout: classic-docs
+:page-liquid:
+:page-description: <<Functionality to optimize running tests on CircleCI>>
+:icons: font
+:toc: macro
+:toc-title:
+
+[#motivation-and-introduction]
+== Motivation and introduction
+
+CircleCI is releasing new preview functionality to **re-run failed tests only**. When selecting this option (see image below), only a subset of tests are re-run, instead of re-running the entire test suite when a transient test failure arises.
+
+Historically, when your testing job in a workflow has flaky tests, the only option to get to a successful workflow was to link:https://support.circleci.com/hc/en-us/articles/360050303671-How-To-Rerun-a-Workflow[re-run your workflow from failed]. This type of re-run executes *all tests* from your testing job, including tests that passed, which prolongs time-to-feedback and consumes credits unneccessarily.
+
+This re-run failed tests only re-runs failed tests from the _same_ commit, not new ones.
+
+image::{{site.baseurl}}/assets/img/docs/rerun-failed-tests-option.png[Option to rerun failed tests from Rerun menu]
+
+[#prerequisites]
+== Prerequisites
+
+* Your testing job in the workflow is configured to xref:collect-test-data/#[upload test results] to CircleCI. `filename` or `classname` attributes **must be present** in the xref:use-the-circleci-cli-to-split-tests#junit-xml-reports[JUnit XML output].
+* The testing job uses `circleci tests run` (see below for details) to execute tests.
++
+NOTE: If your current job is using xref:test-splitting-tutorial#[intelligent test splitting], you must change the `circleci tests split` command to `circleci tests run` (see below for instructions).
+
+[#quickstart]
+== Quickstart
+
+[#example-config-file-before]
+=== Before: Example `.circleci/config.yml` file
+
+```yaml
+ - run:
+    name: Run tests
+    command: |
+      mkdir test-results
+      TEST_FILES=$(circleci tests glob "**/test_*.py" | circleci tests split --split-by=timings) \
+      pytest --junitxml=test-results/junit.xml $TEST_FILES
+      
+- store_test_results
+    path: test-results
+```
+
+This example snippet is from a CircleCI configuration file that:
+
+. Executes Python test files that end in `.py`, 
+. Splits tests by previous timing results (you can follow xref:test-splitting-tutorial#[this tutorial] on intelligent test splitting), 
+. Stores the test results in a new directory called `test-results`, and 
+. Uploads those test results to CircleCI.  
+
+**Note:** `-o junit_family=legacy` is present to ensure that the test results being generated contain the `filename` attribute. Not included in this snippet is the key to set parallelism (read the xref:parallelism-faster-jobs#[Test splitting and parallelism] page for more information).
```

Comment:
Is the `-o junit_family=legacy` supposed to go somewhere in the command?

Reply:
sorry yes, the command on line 45 should be:

`pytest -o junit_family=legacy --junitxml=test-results/junit.xml $TEST_FILES`